### DISCUSSION TITLE
View that references another view

### DISCUSSION THREAD

**Discussion by lukecarr** [planetscale/discussion#420] [Bug Reports] [answered]
I'm sorry in advance if this is a MySQL-centric issue (I'm much more familiar with Postgres and MSSQL, so I could be missing something important here).

I'm creating a view in a development branch (and merging it to my main prod branch). The view references (queries) another view that already exists.

I can successfully create the new view on the dev branch, but when I go to make a deploy request, it fails with an error of "View references non-existing table". As mentioned, I can create and use this view on the dev branch, so there are no issues there (I know for a fact that the view being referenced already exists); it only crops up when trying to merge to main.

**Comment by jgreet** [comment, marked as answer]
Hi @lukecarr,

MySQL supports nested views, and PlanetScale should as well, but as you noticed it's currently impossible to deploy a change containing nested views or promote a branch already containing nested views to "production" status. We're working on a fix and I'll drop an update here when it's rolled out.

For the moment, the only workaround for this bug is to rewrite the views that reference other views.

**Reply by jgreet** [reply]
Promoting and deploying nested views should be working again now. Let us know if you're still having any trouble with them.

**Reply by lukecarr** [reply]
Sorry for the late reply. All working here now! 🎉 Thanks for the quick turnaround on resolving this.
